Update the consul-k8s cli docs for the new `proxy log` subcommand (#16458)
hashicorp/consul: website/content/docs/k8s/k8s-cli.mdx

| ---------------------------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ | ---------------------------------------------------------------------------------------------------------------------- |
| <nobr>`-namespace`, `-n`</nobr>    | `String` Specifies the namespace containing the target Pod.                                                                                                                                                                                                                                                              | Current [kubeconfig](https://kubernetes.io/docs/concepts/configuration/organize-cluster-access-kubeconfig/) namespace. |

> | <nobr>`-update-level`, `-u`</nobr> | `String`  Specifies the logger (optional) and the level to update. <p>Use the following format to configure the same level for loggers: `-update-level <LEVEL>`.</p> <p>You can also specify a comma-delineated list to configure levels for specific loggers, for example: `-update-level grpc:warning,http:info`.</p>  | none                                                                                                                   |

Author: @trujillo-adam what do you think of the text here? I modified it slightly from your suggestion to show that they can either blanket update all the level for all the loggers or target specific ones

of a Kubernetes Pod. If a Pod is running more than one Envoy proxy managed by
Consul, as in the [Multiport configuration](/consul/docs/k8s/connect#kubernetes-pods-with-multiple-ports),
configuration for all proxies in the Pod will be displayed.

The following options are available.

| Flag                               | Description                                                                                                                                                                                                                                                               | Default                                                                                                                |
| ---------------------------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ---------------------------------------------------------------------------------------------------------------------- |

> | <nobr>`-namespace`, `-n`</nobr>    | `String` The namespace where the target Pod can be found.                                                                                                                                                                                                                 | Current [kubeconfig](https://kubernetes.io/docs/concepts/configuration/organize-cluster-access-kubeconfig/) namespace. |

Author: as for the data type, this was following some of the other entries in this file

> | <nobr>`-update-level`, `-u`</nobr> | `String` Update the level for the logger. Can be either `-update-level <LEVEL>` to change all loggers to a specific level, or a comma delineated list of loggers with level can be passed like `-update-level grpc:warning,http:info` to only modify specific loggers.    | none                                                                                                                   |

Author: > Is "update level" a clear or standard type of logger?

I'm not entirely sure what you're asking, `update-level` is the option here that is used to modify the log level, there is no default here because the user is specifying a level to make a change to (so for example "I want to set the grpc logger to warning" or "I want to change all loggers to debug"). The `reset` command has a default because that is what it changes all log levels

| <nobr>`-reset`, `-r`</nobr>        | `String` Reset the log levels for all loggers back to the default of `info`                                                                                                                                                                                               | `info`                                                                                                                 |

#### Example commands

> Get the log levels for all loggers of an Envoy proxy in a given pod.
> `server-697458b9f8-4vr29`.

Reviewer: ```suggestion
In the following example, Consul returns the log levels for all of an Envoy proxy's loggers in a pod with the ID `server-697458b9f8-4vr29`:
```
The ID referenced as part of the example should be a part of the sentence.

connection                      info
health_checker                  info
kafka                           info
mongo                           info
config                          info
admin                           info
forward_proxy                   info
misc                            info
websocket                       info
dubbo                           info
happy_eyeballs                  info
main                            info
client                          info
lua                             info
udp                             info
cache_filter                    info
filter                          info
multi_connection                info
quic_stream                     info
router                          info
http2                           info
key_value_store                 info
secret                          info
testing                         info
upstream                        info
assert                          info
ext_authz                       info
rbac                            info
decompression                   info
envoy_bug                       info
file                            info
pool                            info
```

> Update the log levels for all loggers of an Envoy proxy in a given pod.

Reviewer: ```suggestion
The following command updates the log levels for all loggers of an Envoy proxy to `warning`.
```
Just to be a little more specific

```shell-session

> $ consul-k8s proxy log server-697458b9f8-4vr29 -u warning

Reviewer: ```suggestion
$ consul-k8s proxy log server-697458b9f8-4vr29 -update-level warning
```
Per the style guide, we should use full flag names.

mongo                           warning
router                          warning
thrift                          warning
grpc                            warning
matcher                         warning
hc                              warning
multi_connection                warning
wasm                            warning
dubbo                           warning
filter                          warning
upstream                        warning
backtrace                       warning
connection                      warning
io                              warning
main                            warning
happy_eyeballs                  warning
rds                             warning
tap                             warning
envoy_bug                       warning
rocketmq                        warning
file                            warning
forward_proxy                   warning
stats                           warning
health_checker                  warning
lua                             warning
secret                          warning
quic_stream                     warning
testing                         warning
websocket                       warning
```

> Update the log levels for specific loggers of an Envoy proxy in a given pod.

Reviewer: ```suggestion
The following command updates the `grpc` log level to `error`, the `http` log level to `critical`, and the `runtime` log level to `debug` for pod ID `server-697458b9f8-4vr29`.
```
We can probably leave off the bit about the pod ID because the important information in the example is demonstrating the syntax for updating specific loggers.

```shell-session

> $ consul-k8s proxy log server-697458b9f8-4vr29 -u grpc:error,http:critical,runtime:debug

Reviewer: ```suggestion
$ consul-k8s proxy log server-697458b9f8-4vr29 -update-level grpc:error,http:critical,runtime:debug
```

rocketmq                        info
hystrix                         info
oauth2                          info
redis                           info
backtrace                       info
file                            info
forward_proxy                   info
kafka                           info
config                          info
router                          info
runtime                         debug
testing                         info
happy_eyeballs                  info
ext_proc                        info
init                            info
lua                             info
health_checker                  info
misc                            info
envoy_bug                       info
jwt                             info
main                            info
quic                            info
upstream                        info
websocket                       info
client                          info
decompression                   info
mongo                           info
multi_connection                info
rds                             info
secret                          info
admin                           info
dubbo                           info
```

> Reset the log levels for all loggers of an Envoy proxy in a given pod back to the default level of `info`.
> ```shell-session
> $ consul-k8s proxy log server-697458b9f8-4vr29 -r

Reviewer: ```suggestion
The following command resets the log levels for all loggers of an Envoy proxy in pod `server-697458b9f8-4vr29` to the default level of `info`.
```shell-session
$ consul-k8s proxy log server-697458b9f8-4vr29 -reset
```
